CMSgov/design-system
# [WNMGDS-511] Fix lint scripts for Child DS

### About
eslint wasnt working for child DSs, because [eslint plugins need to be installed relative to the project](https://eslint.org/docs/developer-guide/shareable-configs#publishing-a-shareable-config) using `@cmsgov/design-system-scripts`. Ordinarily this would be done by defining every eslint plugin as a peer dep. However in my testing I found that eslint dependencies just work when you install them via an eslint config package (i.e. `eslint-config-nava` or the original `@cmsgov/eslint-config-design-system`).

Because I didn't want to add 5-7 eslint plugins as peer deps, and I was also concerned about the child DS's ability to customize their linting, I decided to bring back the eslint and stylelint config packages.

The implication is that linting in child design systems will work more similarly to how they currently work in site packages. Instead of providing a linting script that handles all the configs, contains all the needed dependencies, and just works, the new linting script will be pretty minimal, and just use the prettier, stylelint & eslint configs and versions defined in the project. We will provide a set of recommended base configs in the form of `@cmsgov/eslint-config-design-system` & `@cmsgov/stylelint-config-design-system` (we don't provide a `prettier` config because the only option we have is using single quotes). 

TLDR: the updated linting script is just a nicer, shorthand way to run prettier, stylelint, and eslint in one go. We provide recommended eslint and stylelint configs packages, but its up the child DS to use them and define their configs.

### Summary
- Restore `@cmsgov/eslint-config-design-system`
- Restore `@cmsgov/stylelint-config-design-system`
- Update `@cmsgov/design-system-scripts`
   - move `eslint`, `stylelint`, `prettier` to peer deps
   - Simplify lint scripts by relying on default config file functionality, rather than handling it all inside `@cmsgov/design-system-scripts`

Here's what a child DS will look like after this PR:
<img width="763" alt="Screen Shot 2020-06-17 at 3 59 07 PM" src="https://user-images.githubusercontent.com/5424713/84949879-8aa62400-b0b3-11ea-9cd7-17dc3b3bbf07.png">


### How to test
- Run `yarn lint` on the main repo to ensure its working the same
- Run `yarn lint` on [child-design-system-example.zip](https://github.com/CMSgov/design-system/files/4787062/child-design-system-example.zip) and ensure it is working
- Compare to running `yarn lint` with the master, and you should get [this error](https://github.com/CMSgov/design-system/pull/672#issuecomment-618422317)
